### ISSUE TITLE
release a new version on crate.io to support arch build

### ISSUE THREAD

**Issue by Thrimbda** [shadowsocks/shadowsocks-rust#150]
I got a build failed under `4.19.20-1-MANJARO`

after some search I found it was caused by the low version of openssl, states [here](https://github.com/rust-lang/rustup.rs/issues/1518#issuecomment-427581471) as rust-lang/rustup.rs#1518
due to the issue, upgrade openssl in our dependencies may resolve the problem. 
```bash
error: failed to run custom build command for `openssl v0.9.24`
process didn't exit successfully: `/tmp/cargo-installZMcAVY/release/build/openssl-699d2aaa11397079/build-script-build` (exit code: 101)                                                                                                       
--- stderr
thread 'main' panicked at 'Unable to detect OpenSSL version', /home/thrimbda/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-0.9.24/build.rs:16:14                                                                                    
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: failed to compile `shadowsocks-rust v1.6.12`, intermediate artifacts can be found at `/tmp/cargo-installZMcAVY`

Caused by:
  build failed
```

**Comment by zonyitoo** [comment]
The current 1.7.0 version involves too many changes, which is not very stable for daily use. That's why I didn't release it to crates.io.

**Comment by zonyitoo** [comment]
Released v1.7.0 to crates.io.
Please check if the latest release have fixed your issue.

**Comment by Thrimbda** [comment]
cool, build succeeded
